metaseq/trainer.py

```diff
@@ -966,6 +970,19 @@ def clip_grad_norm(
             skip_gradient_update_on_clip_norm=skip_gradient_update_on_clip_norm,
         )
 
+    def skip_spike(self, logging_outputs, max_loss_to_skip_batch):
+        if max_loss_to_skip_batch == -1:
+            return
+        loss_sum = sum(log.get("loss", 0) for log in logging_outputs)
+        sample_size = sum(log.get("sample_size", 0) for log in logging_outputs)
+        loss = float(loss_sum / sample_size / math.log(2))
```

Comment:
This is the same calculation that happens for logging in reduce_metrics: https://github.com/facebookresearch/metaseq/blob/37809ec65632c434ed722a5c8aa37cd194ce9f8b/metaseq/criterions/cross_entropy.py#L103
The loss value used here matches the one that is logged.